runtime/lua/vim/lsp/completion.lua

```diff
@@ -154,8 +154,13 @@ local function get_completion_word(item)
       return item.label
     end
   elseif item.textEdit then
-    local word = item.textEdit.newText
-    return word:match('^(%S*)') or word
+    local word = item.textEdit.newText:gsub('%s+$', '')
+    if word:match('^%W') then
+      local matched_word = vim.fn.matchstr(word, '\\k\\+')
```

Comment:
It's often useful to show a short input/output example
```suggestion
      -- "..." => "..."
      local matched_word = vim.fn.matchstr(word, '\\k\\+')
```